wolfcrypt/src/port/Espressif/esp32_sha.c

```diff
@@ -143,40 +174,50 @@ static const char* TAG = "wolf_hw_sha";
             return _sha_lock_count;
         }
 
-        void* esp_sha_mutex_ctx_owner(void)
-        {
-            void* ret = 0;
-            taskENTER_CRITICAL(&sha_crit_sect);
-            {
-                ret = mutex_ctx_owner;
-            }
-            taskEXIT_CRITICAL(&sha_crit_sect);
-            return ret;
-        };
-    #else
-        int esp_sha_mutex_ctx_owner(void)
-        {
-            return (int)sha_mutex;
-        }
     #endif
 #endif
 
+/* esp_set_hw - set hardware lock, but only if there's no other known
+ * current mutex owner. */
+int esp_set_hw(WC_ESP32SHA* ctx)
+{
+    int ret = ESP_FAIL;
+    if ((uintptr_t)ctx == mutex_ctx_owner || mutex_ctx_owner == NULLPTR) {
+        ESP_LOGV(TAG, "Initializing current mutext owner!");
+        if (esp_sha_hw_islocked(ctx)) {
+             ESP_LOGV(TAG, "esp_set_hw already locked: 0x%x", (intptr_t)ctx);
+        }
+        ctx->mode = ESP32_SHA_HW;
+        mutex_ctx_owner = (uintptr_t)ctx;
+        ret = ESP_OK;
+    }
+    else {
+        ESP_LOGV(TAG, "esp_sha_init_ctx HW for non-owner 0x%x", (intptr_t)ctx);
+    }
+    return ret;
+}
+
 /*
 ** The wolfCrypt functions for LITTLE_ENDIAN_ORDER typically
 ** reverse the byte order. Except when the hardware doesn't expect it.
 **
+** For SoC devices with no HW (Hardware Acceleration) support:
+**    ctx->sha_type will be SHA_INVALID
+**    ctx->mode     will be ESP32_SHA_SW
+**
 ** Returns 0 (FALSE) or 1 (TRUE); see wolfSSL types.h
 */
 int esp_sha_need_byte_reversal(WC_ESP32SHA* ctx)
 {
-    int ret = TRUE; /* assume we'll need reversal, look for exceptions */
+    int ret = TRUE; /* Assume we'll need reversal, look for exceptions. */
```

Comment:
Can you avoid using TRUE/FALSE? Prefer 1=true, 0=false.

Reply:
Yes, no problem. I personally see the TRUE / FALSE as more readable, but the 1=true, 0=false is probably more common.